packages/terra-time-input/src/TimeInput.jsx

```diff
@@ -216,6 +224,24 @@ class TimeInput extends React.Component {
     });
   }
 
+  /* When Time-Input is used with date-time-picker time-input In missing hour scenario Time-Input skips state update due to
+     prevProps and props comparsion. To handle this scenario propValue is being validated with prevState values in below method
+     to ensure state update happens on time value change.
+  */
+  hasTimeValueUpdated(prevState) {
+    let hour = TimeUtil.splitHour(this.props.value);
+    if (this.props.variant === TimeUtil.FORMAT_12_HOUR) {
+      hour = TimeUtil.parseTwelveHourTime(hour, this.anteMeridiem, this.postMeridiem).hourString;
```

Comment:
A check for a valid `hour` should be added here before calling `parseTwelveHourTime` as an empty or undefined hour would return NaN. See [here](https://github.com/cerner/terra-framework/blob/Update-Missing-hour/packages/terra-time-input/src/TimeInput.jsx#L155-L162) for an example.

Reply:
Since all this check will still happens even if we skip update. removed this method and moved this code to [componentDidUpdate() ](9c2a5b2eb31f2d3ca94033865f243793f5add4cc)

Reply:
@supreethmr I have two other ideas that could potentially handle this whole scenario but I wanted to test it first and will update you.

Reply:
I have simplified the comparisons of props value to update state and also I have added condition to update state only when props gets updated : 7d1a4a89def749924df1b0ac06c1ae5887a71939

Reply:
@supreethmr I made some [changes](https://github.com/cerner/terra-framework/compare/master...force-update-missing-hour) here that you can take a look and test to see how this works. The changes in date-time-picker basically allows the time-input to render the missing hour. Then in `componentDidUpdate` if the time is the missing hour it will update the time value and force a render with the correct time.

Reply:
Thank you @benbcai  Changes works fine I have updated the code :  94624440855ecca7249f9c3cd35ae80cda292300